flink-core/src/main/java/org/apache/flink/api/common/state/StateDescriptor.java

```diff
@@ -103,8 +103,13 @@
     /** The configuration of state time-to-live(TTL), it is disabled by default. */
     @Nonnull private StateTtlConfig ttlConfig = StateTtlConfig.DISABLED;
 
-    /** The default value returned by the state when no other value is bound to a key. */
-    @Nullable protected transient T defaultValue;
+    /**
+     * The default value returned by the state when no other value is bound to a key.
+     *
+     * @deprecated since user cannot judge whether the returned default value is for the real
+     *     current state value or just itself, which may make the semantics unclear.
```

Comment:
I think the description might be better:
`To make the semantics more clear, user should manually manage the default value if the contents of the state is {@code null}.`